packages/macros/src/babel/module-exists.ts

```diff
@@ -15,7 +14,7 @@ export default function moduleExists(path: NodePath<t.CallExpression>, state: St
     throw error(assertArray(path.get('arguments'))[0], `the first argument to moduleExists must be a string literal`);
   }
   try {
-    resolve.sync(moduleSpecifier.value, { basedir: dirname(state.sourceFile) });
+    require.resolve(moduleSpecifier.value, { paths: [process.cwd(), state.sourceFile, dirname(state.sourceFile)] });
```

Comment:
Why would this not be a bug in resolve, considering this change?

Do you have a smaller-than-ember-data repro repo?

Reply:
@NullVoxPopuli this definitely would not be a bug in resolve, resolve is doing the correct thing here. The trouble is that `basedir` here made the assumption that the node_modules for `sourceFile` was nested within the node_modules for the project; however, there are quite a large number of situations in which this might not be true even with `yarn` and `npm`, which is why even though I suspect in *this* instance that `pnpm` created an incorrect node_modules heirarchy this is a good change for us to make here.

To explain why it's not a bug in resolve, here is the structure resolve is seeing. In this scenario, `::Store` is a hardlink to  the `Store` package within EmberData.


```
<project>
    // I'm not sure yet exactly how pnpm determines which hash-groups
    // to build and how to point main at one, but it
    // points at one that has correctly assembled the
    // available peer-deps for store
|_node_modules/.pnpm/file+packages+store+hash/node_modules/@ember-data
|                                                   |_ ::Store
|                                                   |_ ::Model
|                                                   |_ ::RecordData
    // however it also points at one that has not correctly
    // assembled the available peer-deps for model
|_node_modules/.pnpm/file+packages+model+hash/node_modules/@ember-data
|                                                   |_ ::Store
|                                                   |_ ::Model
|_ tests/main/node_modules/@ember-data
                                                   |_ ::Store
                                                   |_ ::Model
                                                   |_ ::RecordData
```